### PR TITLE
New Brunswick Child Tax Benefit

### LINKED ISSUE #202: New Brunswick child tax benefit
[Reference](https://www2.gnb.ca/content/gnb/en/departments/finance/taxes/child_tax_benefit.html#:~:text=The%20New%20Brunswick%20Child%20Tax,under%20the%20age%20of%2018.)

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #202

### CODE REVIEW

File: policyengine_canada/parameters/gov/provinces/nb/benefits/benefits.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+description: New Brunswick list of benefits.
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/nb/benefits/benefits.yaml:1]:
```suggestion
description: New Brunswick provides these benefits.
```

File: policyengine_canada/parameters/gov/provinces/nb/benefits/benefits.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+values:
+  2022-01-01:
+    - nb_child_benefit
+
+metadata:
+  unit: program
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/nb/benefits/benefits.yaml:7]:
```suggestion
  unit: program
  label: New Brunswick benefits
```

File: policyengine_canada/parameters/gov/provinces/nb/benefits/nbcb/base.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+description: New Brunswick Child Tax Benefit program base
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/nb/benefits/nbcb/base.yaml:1]: Similar phrasing and punctuation for other descriptions please
```suggestion
description: New Brunswick provides this base amount for its Child Tax Benefit.
```

File: policyengine_canada/parameters/gov/provinces/nb/benefits/nbcb/multiple_children_phase_out/rate.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+description: New Bushwick tax benefit multiple children phase out rate
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/nb/benefits/nbcb/multiple_children_phase_out/rate.yaml:1]: And in other files
```suggestion
description: New Brunswick tax benefit multiple children phase out rate
```

File: policyengine_canada/parameters/gov/provinces/nb/benefits/nbcb/multiple_children_phase_out/rate.yaml
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/nb/benefits/nbcb/multiple_children_phase_out/rate.yaml:1]: Could you reorganize this into `phase_out/{one_child,multiple_children}.yaml`?

File: policyengine_canada/variables/gov/provinces/nb/benefits/nbcb/nb_child_benefit.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    definition_period = YEAR
+    defined_for = ProvinceCode.NB
+
+    def formula(household, period, parameters):
+        income = household("adjusted_family_net_income", period)
+        children = household("nb_child_benefit_eligible_children", period)
+        p = parameters(period).gov.provinces.nb.benefits.nbcb
+        base = p.base * children
+        supplement = household("nb_child_benefit_supplement", period)
+        return (
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/nb/benefits/nbcb/nb_child_benefit.py:17]: Suggested breaking up for clarity.

Also to clarify - can households without children receive the supplement? The current logic suggests they can unless I'm misreading.
```suggestion
        eligible = children > 0
        reduction = where(children > 1, p.multiple_children_phase_out.rate.calc(income), p.one_child_phase_out.rate.calc(income))
        phased_out_base = max_(base - reduction, 0)
        return phased_out_base + supplement
```

File: policyengine_canada/variables/gov/provinces/nb/benefits/nbcb/nb_child_benefit_eligible_child.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+class nb_child_benefit_eligible_child(Variable):
+    value_type = bool
+    entity = Person
+    label = "New Brunswick family benefit eligible child"
+    definition_period = YEAR
+    defined_for = ProvinceCode.NB
+
+    def formula(person, period, parameters):
+        age = person("age", period)
+        return (
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/nb/benefits/nbcb/nb_child_benefit_eligible_child.py:13]: Also just checking this is < not <=
```suggestion
        p = parameters(period).gov.provinces.nb.benefits.nbcb
        return age < p.eligible_age
```

File: policyengine_canada/variables/gov/provinces/nb/benefits/nbcb/nb_child_benefit_supplement.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+    def formula(household, period, parameters):
+        income = household("adjusted_family_net_income", period)
+        p = parameters(
+            period
+        ).gov.provinces.nb.benefits.nbcb.working_income_supplement
+        working_income = household("family_working_income", period)
+        return min_(
+            p.base,
+            max_(
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/nb/benefits/nbcb/nb_child_benefit_supplement.py:19]: Could you break this statement up and add a comment? Seems a bit complex.

**Review by MaxGhenis — COMMENTED** [review]
Also run `make format`

File: policyengine_canada/variables/gov/provinces/nb/benefits/nbcb/nb_child_benefit_eligible_child.py (hunk 6 above)
Comment (PavelMakarchuk) [reply policyengine_canada/variables/gov/provinces/nb/benefits/nbcb/nb_child_benefit_eligible_child.py:13]: Confirmed

File: policyengine_canada/variables/gov/provinces/nb/benefits/nbcb/nb_child_benefit.py (hunk 5 above)
Comment (PavelMakarchuk) [reply policyengine_canada/variables/gov/provinces/nb/benefits/nbcb/nb_child_benefit.py:17]: Yes, the working supplement is applied to families with working income regardless of the number or existence of children.

**Pull request merged**